src/store/index.js

```diff
@@ -13,145 +13,166 @@ import remoteConnection from './plugins/remoteConnection'
 import notificationOnRemoteConnection from './plugins/notificationOnRemoteConnection'
 import decryptAccounts from './plugins/decryptAccounts'
 import aeppApi from './plugins/aeppApi'
+import promisifySecureStorage from '../lib/promisifySecureStorage'
 
 Vue.use(Vuex)
 
-const store = new Vuex.Store({
-  strict: process.env.NODE_ENV !== 'production',
-  plugins: [
-    createPersistedState({
-      paths: [
-        'peerKey',
-        ...IS_MOBILE_DEVICE
-          ? [
-            'apps',
-            'rpcUrl',
-            'mobile.keystore',
-            'mobile.accountCount',
-            'selectedIdentityIdx',
-            'addressBook',
-            'mobile.followers',
-            'mobile.names'
-          ] : []
-      ],
-      setState: (key, state, storage) =>
-        storage.setItem(key, JSON.stringify(state, (key, value) =>
-          value instanceof ArrayBuffer
-            ? { type: 'ArrayBuffer', data: Array.from(new Uint8Array(value)) }
-            : value)),
-      getState: (key, storage) =>
-        JSON.parse(storage.getItem(key), (key, value) =>
-          value && value.type === 'ArrayBuffer'
-            ? new Uint8Array(value.data).buffer
-            : value)
-    }),
-    pollBalance,
-    initEpoch,
-    remoteConnection.plugin,
-    aeppApi,
-    ...IS_MOBILE_DEVICE
-      ? [decryptAccounts, notificationOnRemoteConnection] : []
-  ],
+export default function (storage) {
+  /**
+   * Object placeholder
+   */
+  let placeholder = {}
 
-  modules: IS_MOBILE_DEVICE ? { mobile } : { desktop },
+  /**
+   * check which instance is available
+   */
+  if (window.SecureStorage && storage instanceof window.SecureStorage) {
+    Object.assign(placeholder, {
+      storage: promisifySecureStorage(storage),
+      asyncStorage: true
+    })
+  } else {
+    Object.assign(placeholder, { storage })
+  }
 
-  state: {
-    peerKey: uuid(),
-    selectedAppIdxToRemove: -1,
-    selectedIdentityIdx: 0,
-    showIdManager: false,
-    balances: {},
-    addresses: [],
-    rpcUrl: networksRegistry[0].url,
-    epoch: null,
-    networkId: null,
-    notification: null,
-    apps: Object.keys(appsRegistry),
-    addressBook: []
-  },
+  /**
+   * Return the Vuex Instance
+   */
+  return new Vuex.Store({
+    strict: process.env.NODE_ENV !== 'production',
+    plugins: [
+      (new VuexPersistence(Object.assign(placeholder, {
+        modules: [
```

Comment:
`modules` are [not preserving nested state fields](https://github.com/championswimmer/vuex-persist/blob/4e7ed1f53861a7f11a4a05d79f0d2352a0d54a48/src/index.ts#L82), `reducer` should be used instead